api/handlers.go

```diff
@@ -5,42 +5,193 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
-	"os"
-	"path/filepath"
+	"strings"
 
-	"github.com/qri-io/dataset"
 	"github.com/qri-io/qri/api/util"
 	"github.com/qri-io/qri/base/archive"
+	"github.com/qri-io/qri/dsref"
 	"github.com/qri-io/qri/lib"
-	"github.com/qri-io/qri/profile"
-	reporef "github.com/qri-io/qri/repo/ref"
 )
 
 // GetHandler is a dataset single endpoint
 func GetHandler(inst *lib.Instance, routePrefix string) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		if !(r.Method == http.MethodGet || r.Method == http.MethodPost) {
+		if r.Method != http.MethodGet {
 			util.NotFoundHandler(w, r)
 			return
 		}
-		params := &lib.GetParams{}
 
-		err := UnmarshalParams(r, params)
+		var err error
+		returnCase, params, err := getCaseAndParamsFromRequest(r)
 		if err != nil {
 			util.WriteErrResponse(w, http.StatusBadRequest, err)
 			return
 		}
-		got, _, err := inst.Dispatch(r.Context(), "dataset.get", params)
+
+		switch returnCase {
+
+		case "zip":
+			outBytes, err := inst.Dataset().GetZip(r.Context(), params)
+			if err != nil {
+				util.RespondWithError(w, err)
+				return
+			}
+			writeFileResponse(w, outBytes, "dataset.zip", "zip")
+			return
+
+		case "csv":
+			outBytes, err := inst.Dataset().GetCSV(r.Context(), params)
+			if err != nil {
+				util.RespondWithError(w, err)
+				return
+			}
+			writeFileResponse(w, outBytes, "body.csv", "csv")
+			return
+
+		case "text/json":
+			res, err := inst.Dataset().Get(r.Context(), params)
+			if err != nil {
+				util.RespondWithError(w, err)
+			}
+			outbytes, err := json.Marshal(res.Value)
+			if err != nil {
+				util.RespondWithError(w, err)
+			}
+			componentName := params.Selector
+			if componentName == "" {
+				componentName = "dataset"
+			}
+			writeFileResponse(w, outbytes, fmt.Sprintf("%s.json", componentName), "json")
+			return
+
+		case "pretty":
+			res, err := inst.Dataset().Get(r.Context(), params)
+			if err != nil {
+				util.RespondWithError(w, err)
+			}
+			outbytes, err := json.MarshalIndent(res.Value, "", "  ")
+			if err != nil {
+				util.RespondWithError(w, err)
+			}
+			componentName := params.Selector
+			if componentName == "" {
+				componentName = "dataset"
+			}
+			writeFileResponse(w, outbytes, fmt.Sprintf("%s.json", componentName), "json")
+			return
+		}
+
+		res, err := inst.Dataset().Get(r.Context(), params)
 		if err != nil {
 			util.RespondWithError(w, err)
 			return
 		}
-		res, ok := got.(*lib.GetResult)
-		if !ok {
-			util.RespondWithDispatchTypeError(w, got)
+
+		if lib.IsSelectorScriptFile(params.Selector) {
+			util.WriteResponse(w, res.Bytes)
 			return
 		}
-		replyWithGetResponse(w, r, params, res)
+
+		util.WriteResponse(w, res.Value)
+	}
+}
+
+// possible return cases:
+//  - "zip"
+//  - "csv"
+//  - "pretty"
+//  - "text/json"
+//  - ""
+func getCaseAndParamsFromRequest(r *http.Request) (string, *lib.GetParams, error) {
```

Comment:
ty, def going to create an enum for this & rename.

Pointing out the `text/json` discrepancy made me realize something, all of these "OutTypes" are actually file types that we are potentially returning. Otherwise we are returning `application/json`. I thought it would be too confusing to have a "json" as a `Case`, since it could be potentially ambigious (hence the odd "text/json"), but renaming that to `OutFileType` would make "json" as a type less confusing.